x-pack/plugins/event_log/generated/mappings.json

```diff
@@ -267,6 +267,42 @@
                         }
                     }
                 },
+                "alert": {
+                    "properties": {
+                        "rule": {
+                            "properties": {
+                                "execution": {
+                                    "properties": {
+                                        "uuid": {
```

Comment:
I also don't see this field being populated in this PR...should it be?

I am also uncertain what this field should be?

In the [alerts as data spreadsheet](https://docs.google.com/spreadsheets/d/1Cqv0bRx1BgY6VP9PY4FK6kppUEZJ5qMPoFgr5_XQXJI/edit#gid=1775019155), this UUID is described as `Reference to the rule saved object ID`. Is that still accurate? It's just a copy of the `rule.id` field? Is that valuable since we already have `rule.id`?

OTOH in the RFC, it is described as `uuid for identifying an individual rule execution`. If that is the case, might it make more sense outside of `kibana.alert.rule` and in a field that could be shared by the alerting event log docs as well? For example, we have event log docs for `provider:alerting` and `action:execute-start` that indicate when the rule execution started. It might make sense to have a rule execution UUID stored here and carried through every event written during the execution. Maybe `kibana.alerting.execution_uuid`?

Reply:
Correct, this field is not being populated just yet. The work to populate this field is being tracked as part of this issue (https://github.com/elastic/kibana/issues/110135) and this in-progress draft PR (https://github.com/elastic/kibana/pull/113058).

The field you're referencing in the ADD spreadsheet with regards to `Reference to the rule saved object ID` looks to be `kibana.alert.rule.uuid`, whereas this field here is `kibana.alert.rule.execution.uuid` ([one line above it in the spreadsheet](https://docs.google.com/spreadsheets/d/1Cqv0bRx1BgY6VP9PY4FK6kppUEZJ5qMPoFgr5_XQXJI/edit#gid=1775019155&range=108:108)). The goal here was to share the same fieldname between the ADD mapping and event-log mapping to better allow users to pivot between the two, and as you mention above, scope it to the `execution` namespace. I think the only open question here is whether alert instances would get their own UUID, but I don't believe there's been any further discussion with regards to that.

Reply:
@spong Oh I see! Thanks for pointing out the difference between `kibana.alert.rule.uuid` and `kibana.alert.rule.execution.uuid`. There are so many rows in the spreadsheet!

I took a look at the draft PR and it looks like this UUID originates in the alerting task runner (which makes sense). Since it originates there, which is at a rule/framework level, might it make more sense to move it out of the `kibana.alert` object and instead into the `kibana.alerting` level? Also happy to move this discussion into the RFC if it is more appropriate there.

Reply:
Of course, happy to provide context! 🙂 Let's move this to the RFC if that's okay -- in previous conversations with the team and @pmuellr we had settled on matching the AAD namespace instead of `kibana.alerting`, but we can re-visit with the larger team.